src/lib/mathlib/math/Vector.hpp

```diff
@@ -298,6 +298,18 @@ class __EXPORT VectorBase
 	}
 
 	/**
+	 * element by element inversion
+	 */
+	const Vector<N> einv() const {
+		Vector<N> res;
+
+		for (unsigned int i = 0; i < N; i++)
+			res.data[i] = 1 / (data[i] + FLT_EPSILON);
```

Comment:
It doesn't really matter. A math operation should not alter the numeric value. I'm even wondering if NaN isn't the right output here.